powersimdata/data_access/data_access.py

```diff
@@ -49,89 +46,58 @@ def move_to(self, file_name, to_dir, change_name_to=None):
         """
         raise NotImplementedError
 
-    def get_base_dir(self, kind, backup=False):
-        """Get path to given kind relative to instance root
-
-        :param str kind: one of {input, output, tmp}
-        :param bool backup: pass True if relative to backup root dir
-        :raises ValueError: if kind is invalid
-        :return: (*str*) -- the specified path
-        """
-        _allowed = list(_dirs.keys())
-        if kind not in _allowed:
-            raise ValueError(f"Invalid 'kind', must be one of {_allowed}")
-
-        root = self.root if not backup else self.backup_root
-        return self.join(root, *_dirs[kind])
-
-    def match_scenario_files(self, scenario_id, kind, backup=False):
+    def tmp_folder(self, scenario_id):
         """Get path matching the given kind of scenario data
 
         :param int/str scenario_id: the scenario id
-        :param str kind: one of {input, output, tmp}
-        :param bool backup: pass True if relative to backup root dir
         :return: (*str*) -- the specified path
         """
-        base_dir = self.get_base_dir(kind, backup)
-        if kind == "tmp":
-            return self.join(base_dir, f"scenario_{scenario_id}")
-        return self.join(base_dir, f"{scenario_id}_*")
-
-    def copy(self, src, dest, recursive=False, update=False):
-        """Wrapper around cp command which creates dest path if needed
-
-        :param str src: path to original
-        :param str dest: destination path
-        :param bool recursive: create directories recursively
-        :param bool update: only copy if needed
-        """
-        raise NotImplementedError
+        return self.join(server_setup.EXECUTE_DIR, f"scenario_{scenario_id}")
 
-    def remove(self, target, recursive=False, confirm=True):
-        """Wrapper around rm command
+    def copy(self, src, dest):
+        """Copy file to new location
 
-        :param str target: path to remove
-        :param bool recursive: delete directories recursively
-        :param bool confirm: prompt before executing command
+        :param str src: path to file
+        :param str dest: destination folder
         """
-        raise NotImplementedError
+        if self.fs.exists(dest) and self.fs.isdir(dest):
```

Comment:
Not sure I understand, `self.fs` is only define in the `LocalDataAccess` class

Reply:
It also refers to the property defined in `SSHDataAccess`. So it will be defined on any subclass instance, although not enforced by the base class (we could use the abc module to do this at some point).

Reply:
Yeah. What would happen if someone instantiate the `DataClass` and call the `copy` method? Would the the `abc` module prevent that?

Reply:
Yeah, it would raise an error in that case. No one should be instantiating `DataAccess` directly though. I believe using `abc` would prevent that.